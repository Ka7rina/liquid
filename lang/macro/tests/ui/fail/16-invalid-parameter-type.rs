use liquid_lang as liquid;

#[liquid::contract(version = "0.1.0")]
mod noop {
    #[liquid(storage)]
    struct Noop {}

    #[liquid(methods)]
    impl Noop {
        pub fn constructor(&mut self) {}

        pub fn noop(&self, value: f32) {}
    }
}

fn main() {}
