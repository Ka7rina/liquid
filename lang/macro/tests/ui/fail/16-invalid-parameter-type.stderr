error[E0277]: the trait bound `f32: liquid_lang::You_Should_Use_An_Valid_Input_Type` is not satisfied
  --> $DIR/16-invalid-parameter-type.rs:12:21
   |
12 |         pub fn noop(&self, value: f32) {}
   |                     ^^^^^^^^^^^^^^^^^ the trait `liquid_lang::You_Should_Use_An_Valid_Input_Type` is not implemented for `f32`

error[E0277]: the trait bound `f32: liquid_ty_mapping::SolTypeNameLen<_>` is not satisfied
 --> $DIR/16-invalid-parameter-type.rs:3:1
  |
3 | #[liquid::contract(version = "0.1.0")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `liquid_ty_mapping::SolTypeNameLen<_>` is not implemented for `f32`
  |
  = note: required because of the requirements on the impl of `liquid_ty_mapping::SolTypeNameLen<_>` for `(f32,)`
  = note: required by `liquid_ty_mapping::SolTypeNameLen::LEN`
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
